apps/web-giddh/src/app/userDetails/components/subscription/subscription.component.ts

```diff
@@ -0,0 +1,410 @@
+                this.menuOneWidth = 4;
+                this.menuTwoWidth = 12;
+            } else if (state.breakpoints['(min-width: 1536px)']) {
+                this.rowLength = 150;
+                this.sideBarBoxLength = 15;
+                this.overallSummaryTopRow = 6;
+                this.overallSummaryBottomRow = 4;
+                this.sideBarBoxWidth = 4;
+                this.mainContentWidth = 12;
+                this.menuOneWidth = 4;
+                this.menuTwoWidth = 12;
+            }
+        });
+
+        this.translationComplete();
+    }
+
+    ngOnChanges(): void {
+        this.translationComplete();
+    }
+    /**
+     * This function will use for get subscribed companies 
+     *
+     * @memberof SubscriptionComponent
+     */
+    public getCompanies(): void {
+        this.showLoader = true;
+
+        //This service will use for get subscribed companies
+        this.subscriptionService.getSubScribedCompanies().pipe(takeUntil(this.destroyed$)).subscribe((res) => {
+            this.showLoader = false;
+            if (res && res.status === "success") {
+                if (!res.body || !res.body[0]) {
+                    this.isPlanShow = true;
+                } else {
+                    this.store.dispatch(this.subscriptionsActions.SubscribedCompaniesResponse(res));
+                }
+            }
+        });
+    }
+
+    /**
+   * This function will refresh the subscribed companies if move company was succesful and will close the popup
+   *
+   * @param {*} event
+   * @memberof SubscriptionsComponent
+   */
+    public addOrMoveCompanyCallback(event: boolean): void {
+        if (event === true) {
+            this.getCompanies();
+        }
+    }
+
+    /**
+     *This function will open the move company popup
+     *
+     * @param {*} company
+     * @memberof SubscriptionComponent
+     */
+    public openModalMove(company: any, event: any) {
+        event.preventDefault();
+        event.stopPropagation();
+        this.selectedCompany = company;
+        this.dialog.open(this.moveCompany, { height: '50%', width: '40%' });
+    }
+
+    /**
+    * This function will use for destroy on next and complete
+    *
+    * @memberof SubscriptionComponent
+    */
+    public ngOnDestroy(): void {
+        this.destroyed$.next(true);
+        this.destroyed$.complete();
+    }
+
+    /**
+     *  This function will use for select subscriptions
+     *
+     * @param {*} subscription
+     * @memberof SubscriptionComponent
+     */
+    public selectSubscription(subscription: any): void {
+        this.selectedSubscription = subscription;
+    }
+
+    /**
+     *  This function will use for filter plans , withing and expiry in dropdown and searching subscriptions
+     *
+     * @memberof SubscriptionComponent
+     */
+    public filterSubscriptions(): void {
+
+        let subscriptions = [];
+        this.subscriptions = [];
+
+        this.subscriptions$.subscribe(response => {
```

Comment:
Consider using `take(1)` as this method is executed every time filtering is applied